x-pack/plugin/core/src/main/java/module-info.java

```diff
@@ -65,6 +65,8 @@
     exports org.elasticsearch.xpack.core.enrich;
     exports org.elasticsearch.xpack.core.eql;
     exports org.elasticsearch.xpack.core.esql;
+    exports org.elasticsearch.xpack.core.esql.action;
+    exports org.elasticsearch.xpack.core.esql.action.internal; // TODO: qualify to esql when modularized
```

Comment:
What does this mean?

Reply:
When the esql plugin code is modularised, with a module-info, then this export of the `xxx.internal` package will be qualified to the esql plugin. In that way, only code in the esql plugin module will be able to access types in the `xxx.internal` package. e.g. 

```
 exports org.elasticsearch.xpack.core.esql.action.internal to org.elasticsearch.esql;
```

There are other examples of this in the codebase, e.g. https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/module-info.java#L29